### PR TITLE
Cf provision

### PR DESCRIPTION
**Pull request by dliggat**
Provisions the entire stack necessary for email-to-slack.

Still TODO: decide on approach for KMS.

### CODE REVIEW

**Comment by chrislopresto** [comment]
@dliggat Going to need a rebase / squash now that I removed the aws.json file - #9

**Comment by chrislopresto** [comment]
Needs a rebase, which will clarify some config stuff in Gruntfile.js

File: Gruntfile.js [hunk 1]
```diff
@@ -14,15 +12,33 @@ module.exports = function(grunt) {
...
+      default: {
+      }
+    },
+    lambda_package: {
+      default: {
+      }
+    },
+    lambda_deploy: {
+        default: {
+            arn: 'arn:aws:lambda:us-west-2:550196518397:function:ses-incoming-emails-LambdaFunction-GAHJ9I62G4XJ',
```
Comment (chrislopresto) [inline Gruntfile.js]: This should come from `dotenv` (current master) or `config` (blissful future)

File: Gruntfile.js [hunk 2]
```diff
@@ -14,15 +12,33 @@ module.exports = function(grunt) {
...
+            options: {
+                profile: 'dliggat'
+            }
+        }
+    },
+
+    create_stack: {
+      stack:   'ses-incoming-emails',
```
Comment (chrislopresto) [inline Gruntfile.js]: Same as above

File: README.md [hunk 3]
```diff
@@ -36,4 +36,24 @@ NOTE: Must have correct profile specified in the shell, as `lambda_invoke` does
...
+```
+
+### Create the cloudformation stack
+`grunt create_stack`
+
+### Obtain the stack outputs
+`grunt describe_stack`
+
+### Deploy the code to lambda
+Edit the `lambda_deploy.default.arn` in `Gruntfile.js` (TODO: pull this from `state/lambda-stack-outputs.json`).
```
Comment (chrislopresto) [inline README.md]: Ah interesting. I'd love to get this out of the Gruntfile. We can make an async API call or config file read as necessary.

File: Gruntfile.js (hunk 1 above)
Comment (dliggat) [inline Gruntfile.js]: `grunt describe_stack` builds a file that contains this value; it's not known until you create a stack. We should pull it from there - problem is - it's weird to set the config from another file containing config. That's part of the reason I want to reimplement `lambda_deploy`

File: Gruntfile.js (hunk 2 above)
Comment (dliggat) [inline Gruntfile.js]: definitely, this one should go into `default.json`

File: index.js [hunk 4]
```diff
@@ -39,47 +32,30 @@ var postSlackMessage = function(context, message) {
   req.end();
 };
 
-var bucketName = process.env.EMAIL_TO_SLACK_S3_BUCKET_NAME;
+var bucketName = jsonfile.readFileSync('state/lambda-stack-outputs.json')['StackOutputs']['S3Bucket'];
```
Comment (chrislopresto) [inline index.js]: Idea... add a `config` instance that reads from this file. Then just use the `config` APIs.

`awsConfig.get('StackOutputs.S3Bucket')`

File: README.md (hunk 3 above)
Comment (dliggat) [inline README.md]: the problem is how we inject the ARN into lambda_deploy if we go that route. Not clear to me if that's reasonably feasible given how `grunt-aws-lambda` is built

File: index.js (hunk 4 above)
Comment (dliggat) [inline index.js]: I keep going back and forth on this. [This section on config load order](https://github.com/lorenwest/node-config/wiki/Configuration-Files#file-load-order) inclines me against it. This is transitory data (that we nonetheless package up with the deployment), not long-lived config data like the other stuff in `default.json`.

File: README.md (hunk 3 above)
Comment (chrislopresto) [inline README.md]: My thinking is that the `grunt create_stack` should persist the lambda arn in a JSON config file somewhere. The Gruntfile could then use `config` to read it when constructing the params used by the `deploy` task.

Comment (chrislopresto) [inline README.md]: We want to keep this individual config out of the Gruntfile so multiple people can use the project with their own config/credentials.

Comment (dliggat) [inline README.md]: *Totally* agree on having `grunt create_stack` build the stack outputs file. I punted on that earlier, as CloudFormation can take a while before your stack enters the `CREATE_COMPLETE` state. So we'd need to do some sort of poll/sleep loop - I presume that would involve a `setTimeout` but I wasn't 100% sure how to best structure it.

Comment (chrislopresto) [inline README.md]: Awesome. This is all quite cool.

**Comment by dliggat** [comment]
@chrislopresto ready for merge, I think

**Pull request merged**